torchaudio/prototype/functional/_dsp.py

```diff
@@ -306,3 +308,101 @@ def frequency_impulse_response(magnitudes):
     device, dtype = magnitudes.device, magnitudes.dtype
     window = torch.hann_window(ir.size(-1), periodic=False, device=device, dtype=dtype).expand_as(ir)
     return ir * window
+
+
+def _overlap_and_add(waveform, stride):
+    num_frames, frame_size = waveform.shape[-2:]
+    numel = (num_frames - 1) * stride + frame_size
+    buffer = torch.zeros(waveform.shape[:-2] + (numel,), device=waveform.device, dtype=waveform.dtype)
+    for i in range(num_frames):
+        start = i * stride
+        end = start + frame_size
+        buffer[..., start:end] += waveform[..., i, :]
+    return buffer
+
+
+def filter_waveform(
+        waveform: torch.Tensor,
+        kernels: torch.Tensor,
+        delay_compensation: int = -1):
+    """Applies filters along time axis of the given waveform.
+
+    This function applies the given filters along time axis in the following manner:
+
+    1. Split the given waveform into chunks of the number of the given filters.
```

Comment:
A little bit confusing here if chunk number or chunk length is equal to the number of filters.

```suggestion
    1. Split the given waveform into chunks. The number of chunks is equal to the number of given filters.
```